satellite/satellitedb/utils.go

```diff
@@ -28,29 +28,54 @@ type postgresNodeIDList storj.NodeIDList
 
 // Value converts a NodeIDList to a postgres array
 func (nodes postgresNodeIDList) Value() (driver.Value, error) {
-	const hextable = "0123456789abcdef"
-
 	if nodes == nil {
 		return nil, nil
 	}
-	if len(nodes) == 0 {
-		return []byte("{}"), nil
+
+	values := make([][]byte, len(nodes))
+	for i, n := range nodes {
+		values[i] = n.Bytes()
+	}
+
+	return convertToPostgres(values), nil
+}
+
+type postgresArrayList [][]byte
+
+// Value converts an array of byte arrays to a postgres array
+func (values postgresArrayList) Value() (driver.Value, error) {
+	if values == nil {
+		return nil, nil
 	}
 
+	return convertToPostgres(values), nil
+}
+
+func convertToPostgres(values [][]byte) []byte {
+	const hextable = "0123456789abcdef"
+
+	if len(values) == 0 {
+		return []byte("{}")
+	}
+
+	var total int
+	for _, path := range values {
+		total += len(path)
+	}
 	var wp, x int
-	out := make([]byte, 2+len(nodes)*(6+storj.NodeIDSize*2)-1)
+	out := make([]byte, 2+(6*len(values))+(total*2)-1)
```

Comment:
what does this math mean? I don't really understand what it was doing originally or what it is doing now. Could we add a comment explaining it?

Reply:
Reverted the util changes. I missed pg.ByteaArray function